src/Reflection/ReflectionClass.php

```diff
@@ -542,6 +534,35 @@ function (ReflectionClassConstant $constant) : string {
     }
 
     /**
+     * Get an array reflection object of the defined constants in this class.
+     *
+     * @return ReflectionClassConstant[]
+     */
+    public function getReflectionConstants() : array
+    {
+        return array_merge(
+            $this->getImmediateReflectionConstants(),
```

Comment:
The immediate reflection constants have precedence over the inherited ones: please do add a test with same constant names, but different values throughout the inheritance.